### PR TITLE
Use `to_dict` in other places

### PR DESCRIPTION
**Pull request by prisae**
Not sure if this is cleaner or not. What do you think @akaszynski ?

### CODE REVIEW

File: scooby/report.py [hunk 1]
```diff
@@ -277,14 +275,12 @@ def cols(html, version, name, ncol, i):
 
         # ########## Platform/OS details ############
         html += "  <tr>\n"
-        html, i = cols(html, self.system, 'OS', self.ncol, 0)
-        html, i = cols(html, self.cpu_count, 'CPU(s)', self.ncol, i)
-        html, i = cols(html, self.machine, 'Machine', self.ncol, i)
-        html, i = cols(html, self.architecture, 'Architecture', self.ncol, i)
-        if TOTAL_RAM:
```
Comment (akaszynski) [inline scooby/report.py]: Need to keep `TOTAL_RAM`.  How about

```python
keys = ['OS', 'CPU(s)', 'Machine', 'Architecture', 'Environment']
if TOTAL_RAM:
    keys.append('RAM')
for key in keys:
    # ...
```

Comment (prisae) [reply scooby/report.py]: But 'RAM' is only in the `repr_dict()` if `TOTAL_RAM`

Comment (prisae) [reply scooby/report.py]: (this check is now just further down in `to_dict` - or do I miss something?)

Comment (prisae) [reply scooby/report.py]: This is why it has the line `if key in repr_dict`

Comment (akaszynski) [reply scooby/report.py]: Sorry, missed that.

Comment (prisae) [reply scooby/report.py]: No problem at all

**Pull request merged**